sr_port/lv_getslot.c

```diff
@@ -141,7 +141,7 @@ lvTreeNode *lvtreenode_getslot(symval *sym)
 					numElems = p->numAlloc;
 				else
 					numElems = LV_NEWBLOCK_INIT_ALLOC;
-				lvtreenode_newblock(sym, (numElems > MAXINT4) ? MAXINT4 : (numElems * 2));
+				lvtreenode_newblock(sym, (numElems < MAXINT4) ? (numElems * 2) : MAXINT4);
```

Comment:
This seems to have an edge case problem. As soon as numElems is > (MAXINT4 / 2) but is < MAXINT4, this expression is either going to pass a negative value to lvtreenode_newblock() or a wrapped (too small) positive value. Not sure exactly which but think it is the former because numElems is unsigned which I think forces unsigned arithmetic. Yet the second parm to lvtreenode_newblock() is defined as being an int so I'm puzzled by the lack of a type sign conflict warning. Were the parm unsigned, there would still be an issue because again, values of numElems where (MAXINT4 / 2) < numElems < MAXINT4, the value of that expression will be larger than when numElems >= MAXINT4.

Reply:
Thanks for the catch. Now fixed.